conman/routes/utils.py

```diff
@@ -8,16 +8,20 @@ def split_path(path):
 
     A url's sub-paths consist of all substrings ending in / and starting at
     the start of the url.
-    """
-    paths = ['/']
-    path = path.rstrip('/')
 
+    eg: /path/containing/subpaths/ becomes:
+
+        /
+        /path/
+        /path/containing/
+        /path/containing/subpaths/
+    """
+    paths = deque()
+    path = path or '/'
     while path:
-        paths.insert(1, path + '/')
-        path = os.path.split(path)[0]
-        if path == '/':
-            break
-    return paths
+        path = path.rpartition('/')[0]
```

Comment:
haha don't feel bad -- this is the third time I've written this function, and it's only just occurred to me `;)`